src/network/ConnectionRegistry.ts

```diff
@@ -0,0 +1,196 @@
+import {ClientConnection} from './ClientConnection';
+import {ClientOfflineError, IOError, LoadBalancer, UUID} from '../core';
+import {ClientConfig, ConnectionStrategyConfig, ReconnectMode} from '../config';
+
+export enum ClientState {
```

Comment:
It might be a good idea to rename this enum into `ClientConnectionState` or simply `ConnectionState`. This way it will feel more natural to see this enum in `ConnectionRegistry`.